scripts/sanityCheck.js

```diff
@@ -0,0 +1,230 @@
+  })
+
+  log.info('[Check 1/2]: Checking Marketplace sanity against all events')
+  try {
+    await marketplaceCheck(shouldFix)
+  } catch (err) {
+    console.log(err, err.stack)
+  }
+
+  log.info('[Check 2/2]: Checking parcel ownership from -150,-150 to 150,150')
+  await processParcels(shouldFix)
+}
+
+async function marketplaceCheck(fix) {
+  const marketplace = eth.getContract('Marketplace')
+  const events = await getAllEvents(marketplace)
+  const filterParcels = getParcelIds(events)
+
+  const tasks = []
+  for (let d of filterParcels) {
+    tasks.push(checkParcel(d.parcelId, d.lastEvent, marketplace, fix))
+  }
+  console.log('Analyzing', tasks.length, 'cases of parcels')
+  await Promise.all(tasks)
+}
+
+function getAllEvents(marketplace) {
+  return new Promise((resolve, reject) => {
+    marketplace.instance
+      .allEvents({
+        fromBlock: 0,
+        toBlock: 'latest'
+      })
+      .get(function(error, events) {
+        if (error) {
+          return reject(error)
+        } else {
+          return resolve(events)
+        }
+      })
+  })
+}
+
+function getParcelIds(events) {
+  const results = {}
+  for (let event of events) {
+    const str = event.args.assetId
+    if (!str) continue
+    if (event.removed && results[str]) continue
+    results[str] = { parcelId: event.args.assetId, lastEvent: event }
+  }
+  return Object.values(results)
+}
+
+const NULL =
+  '0x0000000000000000000000000000000000000000000000000000000000000000'
+
+async function checkParcel(parcelId, lastEvent, marketplace, fix) {
+  const res = await decodeAssetId(parcelId)
+  const [x, y] = res.split(',')
+  const publication = await Publication.findInCoordinate(x, y)
+  const auction = await marketplace.instance.auctionByAssetId(parcelId)
+
+  if (auction[0] !== NULL) {
+    // Check if the publication exists in db
+    if (!publication.length) {
+      console.log(x, y, 'missing publication in db')
+      if (fix) {
+        await insertPublication(x, y, lastEvent)
+        console.log(x, y, 'publication fixed')
+      }
+      return
+    }
+    // Check that id matches
+    const pub = publication[0]
+    if (pub.contract_id !== auction[0]) {
+      console.log(
+        x,
+        y,
+        'different id in db',
+        pub.contract_id,
+        'vs in blockchain',
+        auction[0]
+      )
+      if (fix) {
+        await deletePublication(x, y)
+        await insertPublication(x, y, lastEvent)
+        console.log(x, y, 'publication fixed')
+      }
+      return
+    }
+  } else if (publication.length) {
+    // Check for hanging publication in db
+    const pub = publication[0]
+    if (pub.status === 'open') {
+      console.log(x, y, 'open in db and null in blockchain')
+      if (fix) {
+        await deletePublication(x, y)
+        console.log(x, y, 'publication fixed')
+      }
+      return
+    }
+  }
+}
+
+function deletePublication(x, y) {
+  return Publication.delete({
+    x,
+    y
+  })
```

Comment:
This scares me. This is potentially removing publications from older owners which might be `cancelled` or `sold` which tampers with our ability to show it on the UI. The method is not really `deletePublication` but `deletePublications`.

Maybe we can use a timestamp or the owner? Do you think this is a problem?